src/context.ts

```diff
@@ -177,60 +179,53 @@ export namespace CheCtlContext {
 
     ctx[EclipseCheContext.CHANNEL] = flags[OLM_CHANNEL_FLAG]
     if (!ctx[EclipseCheContext.CHANNEL]) {
-      if (ctx[CliContext.CLI_IS_DEV_VERSION]) {
-        ctx[EclipseCheContext.CHANNEL] = EclipseChe.NEXT_CHANNEL
-      } else {
-        ctx[EclipseCheContext.CHANNEL] = EclipseChe.STABLE_CHANNEL
-      }
-    }
-
-    if (ctx[CliContext.CLI_IS_CHECTL]) {
-      if (ctx[EclipseCheContext.CHANNEL] === EclipseChe.STABLE_CHANNEL) {
-        ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE] = EclipseChe.STABLE_CATALOG_SOURCE_IMAGE
-      } else {
-        ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE] = EclipseChe.NEXT_CATALOG_SOURCE_IMAGE
-      }
-    } else {
-      if (ctx[EclipseCheContext.CHANNEL] !== 'stable') {
-        let iibImageTag = 'next'
-        if (ctx[EclipseCheContext.CHANNEL] === 'latest') {
-          iibImageTag = 'latest'
-        }
-        // It is always EclipseChe.NEXT_CHANNEL, 'latest` and 'next` are added only to simplify UI
-        // See https://issues.redhat.com/browse/CRW-3877
-        ctx[EclipseCheContext.CHANNEL] = EclipseChe.NEXT_CHANNEL
-        ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE] = `quay.io/devspaces/iib:${iibImageTag}-v${ctx[InfrastructureContext.OPENSHIFT_VERSION]}-${ctx[InfrastructureContext.OPENSHIFT_ARCH]}`
-      }
+      ctx[EclipseCheContext.CHANNEL] = ctx[CliContext.CLI_IS_DEV_VERSION] ? EclipseChe.NEXT_CHANNEL : EclipseChe.STABLE_CHANNEL
+    } else if (!ctx[CliContext.CLI_IS_CHECTL] && ctx[EclipseCheContext.CHANNEL] !== EclipseChe.STABLE_CHANNEL) {
+      // It is always EclipseChe.NEXT_CHANNEL, 'latest` and 'next` are added only to simplify UI
+      // See https://issues.redhat.com/browse/CRW-3877
+      ctx[EclipseCheContext.CHANNEL] = EclipseChe.NEXT_CHANNEL
     }
 
     ctx[EclipseCheContext.PACKAGE_NAME] = flags[PACKAGE_MANIFEST_FLAG] || EclipseChe.PACKAGE
     ctx[EclipseCheContext.CATALOG_SOURCE_NAMESPACE] = flags[CATALOG_SOURCE_NAMESPACE_FLAG] || ctx[InfrastructureContext.OPENSHIFT_MARKETPLACE_NAMESPACE]
+
     ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = flags[CATALOG_SOURCE_NAME_FLAG]
     if (!ctx[EclipseCheContext.CATALOG_SOURCE_NAME]) {
-      if (ctx[EclipseCheContext.CHANNEL] === EclipseChe.STABLE_CHANNEL) {
-        ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = EclipseChe.STABLE_CHANNEL_CATALOG_SOURCE
-      } else {
-        ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = EclipseChe.NEXT_CHANNEL_CATALOG_SOURCE
-      }
+      ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = ctx[EclipseCheContext.CHANNEL] === EclipseChe.STABLE_CHANNEL ? EclipseChe.STABLE_CHANNEL_CATALOG_SOURCE : EclipseChe.NEXT_CHANNEL_CATALOG_SOURCE
+    }
+
+    ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE] = flags[CATALOG_SOURCE_IMAGE_FLAG]
+    if (ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE]) {
+      ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = EclipseChe.NEXT_CHANNEL_CATALOG_SOURCE
```

Comment:
I'm not sure this is correct... for dsc, I could be installing from quay.io/devspaces/iib:3.6.0.RC-04-27-v4.11-x86_64, which is technically the `latest` tag of the `fast` channel. Does `NEXT_CHANNEL_CATALOG_SOURCE` always equal `fast` ?

Reply:
It is `fast` for dsc [1]
https://github.com/redhat-developer/devspaces-chectl/blob/devspaces-3-rhel-8/build/scripts/sync.sh#L95-L106

Reply:
will I be able to use `server:update` to move from fast channel to stable channel, eg., for 

3.5.0.GA (stable) -> 3.6.0.RC (fast) -> 3.6.0.GA (stable)